src/Orleans.Runtime/Hosting/GenericHostExtensions.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using Microsoft.Extensions.Hosting;
+
+namespace Orleans.Hosting
+{
+    public static class GenericHostExtensions
+    {
+        public static IHostBuilder AddOrleans(this IHostBuilder hostBuilder, Action<ISiloBuilder> configureDelegate)
+        {
+            if (configureDelegate == null) throw new ArgumentNullException(nameof(configureDelegate));
+            var siloBuilder = new SiloBuilder(hostBuilder);
```

Comment:
This should probably use either the `Properties` dictionary or `IServiceCollection` - see how `ApplicationParts` does it via `ApplicationPartManagerExtensions.GetApplicationPartManager(builder.Properties)`.

AspNetCore uses `IServiceCollection` for the same thing: https://github.com/aspnet/AspNetCore/blob/321327f84b2b22dcff2e9beb06a9a64236c5cced/src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/MvcCoreServiceCollectionExtensions.cs#L74 << maybe that approach offers some benefits - unclear.

Happy for you to take either route. It ultimately shouldn't matter because the `SiloBuilder` itself is stateless. Maybe in future that will change.